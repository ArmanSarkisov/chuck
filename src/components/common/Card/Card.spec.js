import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Card } from './index';

describe('Card test', () => {
  let component;

  describe('Has props', () => {
    beforeEach(() => {
      component = shallow(<Card text="Some text" categories={['Category']} />);
    });

    it('should render Card component', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render props text', () => {
      const p = component.find('.text-base');
      expect(p.text()).toBe('Some text');
    });

    it('should render props category', () => {
      const span = component.find('.span');
      expect(span).toHaveLength(1);
    });
  });

  describe('Has not props', () => {
    beforeEach(() => {
      component = shallow(<Card />);
    });

    it('should render Card component without props', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render props default text', () => {
      const p = component.find('.text-base');
      expect(p.text()).toBe('Click on button get random text');
    });

    it('should render props default category', () => {
      const span = component.find('.span');
      expect(span).toHaveLength(0);
    });
  });
});
