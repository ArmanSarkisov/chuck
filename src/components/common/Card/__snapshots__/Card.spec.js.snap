// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test Has not props should render Card component without props 1`] = `
<div
  className="max-w-sm rounded overflow-hidden shadow-lg"
>
  <div
    className="px-6 py-4"
  >
    <div
      className="font-bold text-xl mb-2"
    >
      The Coolest Joke
    </div>
    <p
      className="text-gray-700 text-base"
    >
      Click on button get random text
    </p>
  </div>
  <div
    className="px-6 pt-4 pb-2"
  />
</div>
`;

exports[`Card test Has props should render Card component 1`] = `ShallowWrapper {}`;
